cupyx/scipy/sparse/compressed.py

```diff
@@ -33,6 +33,92 @@ class _compressed_sparse_matrix(sparse_data._data_matrix,
         ''',
         'compress_getitem_complex')
 
+    # https: // www.micc.unifi.it / bertini / download
+    # / gpu - programming - basics / 2017 / gpu_cuda_5.pdf
+    setitem_preamble = '''
+            #if __CUDA_ARCH__ < 600
+            __device__ double atomicExch(double* address, double val)
+            {
+                unsigned long long int* address_as_ull =
+                                          (unsigned long long int*)address;
+                unsigned long long int old = *address_as_ull, compare;
```

Comment:
Could you split the declaration of `compare` into two lines? This is a very confusing statement.

Reply:
I think that `uint64_t address_as_ull = reinterpret_cast<uint64_t>(address);` should be enough

Reply:
> Could you split the declaration of `compare` into two lines? This is a very confusing statement.

Do you mean to re-declare the data type ?